src/test/ui/recursion/issue-83150.rs

```diff
@@ -0,0 +1,9 @@
+fn main() {
```

Comment:
This test passes for an unexplainable reason. When I compile this file I get the expected error:

```
error[E0275]: overflow evaluating the requirement `Map<&mut Map<&mut Map<&mut Map<&mut Map<.....
```

But the test suite cannot find an error.